File: MiniProjectManager/Frontend/mini-project-manager-frontend/src/types/project.ts
export interface Project {
    id: number;
    title: string;
    description?: string;
    creationDate: string;
    userId: number;
}

export interface CreateProjectDto {
    title: string;
    description?: string;
}

export interface UpdateProjectDto {
    title: string;
    description?: string;
}

File: MiniProjectManager/Backend/MiniProjectManager.Api/Models/Project.cs
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;

namespace MiniProjectManager.Api.Models
{
    public class Project
    {
        public int Id { get; set; }

        [Required]
        [MinLength(3)]
        [MaxLength(100)]
        public string Title { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }

        public DateTime CreationDate { get; set; }

        public int UserId { get; set; }
        public User User { get; set; } = default!;

        public ICollection<TaskItem> Tasks { get; set; } = new List<TaskItem>();
    }
}


